inginious/frontend/plugins/manual_scoring/static/js/rubric.js

```diff
@@ -139,4 +140,59 @@ class MatrixField {
     get fieldId() {
         return `${this.iIndex}-${this.jIndex}`;
     }
+}
+
+class Score {
+    constructor(textId, scoreValue) {
+        this.text = $(`#${textId}`);
+        this.score = scoreValue;
+    }
+
+    changeColor() {
+        const textColor = this.pickColor(this.score);
+        this.text.css("color", textColor);
+    }
+
+    updateScore() {
+        let currentScore;
+        try {
+            currentScore = parseFloat(this.score).toFixed(1);
+        } catch (error) {
+            currentScore = this.score;
+        }
+        this.text.html(currentScore);
+    }
+
+    pickColor(score) {
+        const defaultText = "No grade";
+        const defaultColor = "#002a95"
+        if (score === defaultText) {
+            return defaultColor;
+        }
+        let color = "";
+        if (score < 0.5) {
```

Comment:
Probably using a dictionary to map the score to a color? or using two arrays and just ask whether the score is within a range or something like that as there are many `if else` statements.